pkg/params/params.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package params provides a generic way to describe parameters used by gadgets, operators
+and runtimes including validation. They can easily be serialized and handed over to different
+frameworks like cobra for use in CLI or a webinterface using JSON.
+*/
+package params
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+type (
+	Params           []*Param
+	ParamMap         map[string]string
+	ParamsCollection map[string]Params
```

Comment:
Do we need all these combinations of types? I'm wondering if this is possible to live only with `Params map[string]*Param` (and maybe `ParamMap map[string]string`)?

I know different combinations bring flexibility, but IMO it also creates confusion and it's not clear what's the difference between them and which one is the right fit for different cases.

Reply:
Hmm, `[]*Param` is what all components return. If we were to change that to `map[string]*Param`, we'd need some kind of automatism that checks whether the key of that map matches the `Key` of the Param - which I'm not a big fan of. If you'd drop the `Key` from Param however, a Param would never be able to live on it's own.

The `.ParamMap()` - which the gadgets right now use to get the config - is there to avoid looping over the array whenever you want to get the value of a parameter by it's key, but maybe that's unnecessary optimization, as there won't be more than a couple of params anyways. Would it help if we drop that and just add a Get(key) to the `Params []*Param`? Sorry for the confusion ;)

Reply:
I think the type `ParamMap` can be deleted, and `ParamMap()` function simply returns `map[string]string`.